solution/ui/regulations/eregs-vite/src/views/Resources.vue

```diff
@@ -251,27 +258,66 @@ export default {
+                    if (payload.searchSection) {
+                        if (!newQueryParams.part.includes(splitSection[0])) {
+                            newQueryParams.part =
+                                newQueryParams.part + "," + splitSection[0];
+                        }
+                    }
+                    const scopeVals = newQueryParams[payload.scope].split(",");
+                    scopeVals.push(payload.selectedIdentifier);
+                    const uniqScopeVals = _uniq(scopeVals);
+                    newQueryParams[payload.scope] = uniqScopeVals
+                        .sort()
+                        .join(",");
+                } else {
+                    newQueryParams.title = "42"; // hard coding for now
+                    if (payload.scope === "section") {
+                        if (newQueryParams.part) {
+                            if (!newQueryParams.part.includes(splitSection[0])) {
+                                newQueryParams.part = newQueryParams.part + "," + splitSection[0];
+                            }
+                        } else {
+                            newQueryParams.part = splitSection[0];
+                        }
+                    }
+                    newQueryParams[payload.scope] = payload.selectedIdentifier;
+                }
+
+                if (payload.scope === "subpart") {
+                    newQueryParams = await this.combineSections(
+                        payload.selectedIdentifier,
+                        newQueryParams
+                    );
+                }
+
+                if (payload.searchSection) {
+                    newQueryParams.q = "";
+                    this.searchInputValue = "";
+                }
 
-            if (newQueryParams[payload.scope]) {
-                const scopeVals = newQueryParams[payload.scope].split(",");
-                scopeVals.push(payload.selectedIdentifier);
-                const uniqScopeVals = _uniq(scopeVals);
-                newQueryParams[payload.scope] = uniqScopeVals.sort().join(",");
-            } else {
-                newQueryParams.title = "42"; // hard coding for now
-                newQueryParams[payload.scope] = payload.selectedIdentifier;
-            }
-            if (payload.scope === "subpart") {
-                newQueryParams = await this.combineSections(
-                    payload.selectedIdentifier,
-                    newQueryParams
-                );
                 this.getPartDict(newQueryParams);
+
+                this.$router.push({
+                    name: "resources",
+                    query: newQueryParams,
+                });
             }
-            this.$router.push({
-                name: "resources",
-                query: newQueryParams,
-            });
+        },
+
+        async checkPart(payload, scope) {
+            const partExist = this.filters.part.listItems.filter((part) => part.name == payload[0]).length > 0;
```

Comment:
.find seems apprpriate here.  you can just replace filter and then lose .length > 0.